src/AutoRest.CSharp.V3/Output/Builders/SerializationBuilder.cs

```diff
@@ -150,24 +150,70 @@ private static XmlObjectSerialization BuildXmlObjectSerialization(ObjectSchema o
                 schemaTypeReference, elements.ToArray(), attributes.ToArray(), embeddedArrays.ToArray());
         }
 
+        private static IEnumerable<JsonPropertySerialization> GetPropertySerializationsFromBag(PropertyBag propertyBag)
+        {
+            foreach (Property property in propertyBag.Properties)
+            {
+                yield return new JsonPropertySerialization(
+                    property.SerializedName,
+                    property.CSharpName(),
+                    BuildSerialization(property.Schema, property.IsNullable()),
+                    false);
+            }
+
+            foreach ((string name, PropertyBag innerBag) in propertyBag.Bag)
+            {
+                JsonPropertySerialization[] serializationProperties = GetPropertySerializationsFromBag(innerBag).ToArray();
+                JsonObjectSerialization objectSerialization = new JsonObjectSerialization(new SchemaTypeReference(null, false), serializationProperties, null);
+                yield return new JsonPropertySerialization(name, name, objectSerialization, true);
```

Comment:
Serialized and source names are the same?

Reply:
Should we make one of the names nullable and remove IsFlattened?

Reply:
Done.